Doc/library/socket.rst

```diff
@@ -277,6 +277,283 @@ Constants
+
+..   Availability: Linux, Windows >= Vista.
+
+.. data:: SO_USELOOPBACK
+
+..   Availability: FreeBSD, Windows >= Vista `(non functional) <https://msdn.microsoft.com/en-us/library/windows/desktop/ms740532(v=vs.85).aspx>`_
+
+.. data:: SO_LINGER
+
+   When a socket is set to linger. A call to :meth:`socket.close` or
+   :meth:`socket.shutdown` will not return until all queued messages
+   for the socket have been successfully sent or the linger timeout
+   is reached. If the socket is closed due to the program exiting, it
+   will linger in the background.
+
+   The value for this option is a struct on most operating systems.
+   Consult your OS documentation for the struct's details.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_OOBINLINE
+
+   Allows receiving of out of band data in the normal stream.
+
+   .. seealso::
+
+      `Out-of-band data <https://www.ibm.com/support/knowledgecenter/en/ssw_i5_54/rzab6/coobd.htm>`_
+      for an in depth explanation.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_REUSEPORT
+
+..   Availability: Linux >= 3.9 (for TCP/UDP)
+
+.. data:: SO_SNDBUF
+          SO_RCVBUF
+
+   Sets the size of the sending and receiving buffers for this socket in
+   bytes. Most operating systems impose an upper limit on the size of
+   these buffers.
+
+..   Availability: Linux, Windows
+
+.. data::
+          SO_RCVLOWAT
+          SO_SNDLOWAT
+
+   SO_RCVLOWAT sets the minimum number of bytes that must be present in
+   the socket's internal receive buffer before they are passed on to able
+   read call. SO_SNDLOWAT similiary sets the minimum bytes before data is
+   sent from the send buffer to the socket protocol.
+
+   SO_SNDLOWAT is read-only on Linux and SO_RCVLOWAT is read-only on
+   Linux versions below 2.4.
+
+   Both these values default to 1.
+
+..   Availability: Linux, Windows `(non functional) <https://msdn.microsoft.com/en-us/library/windows/desktop/ms740532(v=vs.85).aspx>`_
+
+.. data:: SO_RCVTIMEO
+          SO_SNDTIMEO
+
+   Specifies the amount of time send and receive calls for this socket willl
+   block before timing out. The default timeout of zero means that operations
+   will never time out. On Linux this is a `struct timeval`, on Windows this
+   is the time in milliseconds.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_ERROR
+
+   Gets and resets the pending socket error.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_TYPE
+
+   Gets the type of the socket. This type corresponds to the type as
+   defined in the :func:`socket.socket` function. Examples include
+   :const:`SOCK_STREAM` and :const:`SOCK_DGRAM`.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_SETFIB
+
+   Sets the FIB (routing table) for the socket.
+
+   Availability: FreeBSD >= 7.1
+
+   .. versionadded:: 3.1
+
+.. data:: SO_PASSCRED
+          SO_PEERCRED
+
+   Allows for the passing of SCM credentials over unix sockets.
+   See the end of :func:`socket.recvmsg` for details.
```

Comment:
Yeah I'm not familiar with this API either, I'm just going to add an easily searchable keyword of ancillary message and point to recvmsg